tree/treeplayer/inc/TTreeReaderArray.h

```diff
@@ -59,90 +59,159 @@ namespace Internal {
 } // namespace ROOT
 
 template <typename T>
-class TTreeReaderArray: public ROOT::Internal::TTreeReaderArrayBase {
+class TTreeReaderArray : public ROOT::Internal::TTreeReaderArrayBase {
 public:
+   /// Create an array reader of branch "branchname" for TTreeReader "tr".
+   TTreeReaderArray(TTreeReader &tr, const char *branchname)
+      : TTreeReaderArrayBase(&tr, branchname, TDictionary::GetDictionary(typeid(T))) {}
+
+   T &At(std::size_t idx) { return *static_cast<T *>(UntypedAt(idx)); }
+   const T &At(std::size_t idx) const { return *static_cast<T *>(UntypedAt(idx)); }
+   T &operator[](std::size_t idx) { return At(idx); }
+   const T &operator[](std::size_t idx) const { return At(idx); }
+
+   /// Random access iterator to the elements of a TTreeReaderArray
+   class Iterator_t : public std::iterator<std::random_access_iterator_tag, T> {
```

Comment:
`std::iterator` is actually deprecated. I'm not requesting a change - I'm suggesting it in case you think it helps this PR. See e.g. https://stackoverflow.com/questions/43268146/why-is-stditerator-deprecated

Reply:
I saw it was deprecated in c++17 and did not think much of it (I thought something else changed in c++17 that allowed its deprecation). After reading the stackoverflow thread you linked, I removed `std::iterator` and just declared the required nested types myself.